django_filters/filters.py

```diff
@@ -307,11 +307,34 @@ class DurationFilter(Filter):
     field_class = forms.DurationField
 
 
-class ModelChoiceFilter(Filter):
+class QuerySetMixin(object):
+    def get_user(self):
+        try:
+            return self.parent.user
+        except AttributeError:
+            return None
+
+    def get_queryset(self, user):
+        queryset = self.extra.get('queryset')
+
+        if callable(queryset):
+            return queryset(user=user)
```

Comment:
Was thinking the API would be similar to a few of django's model field options, such as [`default`](https://docs.djangoproject.com/en/1.10/ref/models/fields/#django.db.models.Field.default). You either pass the value (qs in this case) or a callable. In this case, the callable is passed the request or user (tbd, but leaning towards passing the request now).